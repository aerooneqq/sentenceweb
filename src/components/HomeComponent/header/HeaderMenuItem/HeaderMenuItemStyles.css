@media screen and (max-width: 560px){ 
    .headerMenuOption {
        width: 100%;
        height: 25%;
        display: flex;
        align-items: center;
        justify-content: center;

        visibility: visible;
    }

        .headerMenuOption:nth-child(1){ 
            background: #141414;
        }

        .headerMenuOption:nth-child(2){ 
            background: #242424;
        }

        .headerMenuOption:nth-child(3){ 
            background: #343434;
        }
        
        .headerMenuOption:nth-child(4){ 
            background: #444444;
        }

    .headerMenuOptionText{        
        font-family: Arial; 
        color: orangered;
        font-size: 19px;
        margin-right: 10px;

        cursor: default;
    }

        .headerMenuOptionText:hover{ 
            color: white;
        }
}

@media screen and (min-width: 560px){ 
    .headerMenuOption {
        width: auto;
        height: 100%;
        background-color: transparent;
        display: flex;
        align-items: center;
        opacity: 0;

        animation-name: menuItemFadeIn;
        animation-duration: 0.5s;
        animation-delay: 0.4s;
        animation-fill-mode: forwards;
    }

    .headerMenuOptionText{
        animation-name: menuOptionTextSlideIn;
        animation-duration: 1s;
        animation-delay: 0.4s;
        animation-fill-mode: forwards;
        
        font-family: Arial; 
        color: orangered;
        font-size: 19px;
        margin-right: 10px;

        cursor: default;
    }
        .headerMenuOptionText:hover{ 
            color: white;
        }

    @keyframes menuItemFadeIn {
        from {
            opacity: 0;
        }
    
        to {
            opacity: 1;
        }
    }

    @keyframes menuOptionTextSlideIn{
        from{
            margin-left: -5px;
        }
    
        to{
            margin-left: 5px;
        }
    }
}






