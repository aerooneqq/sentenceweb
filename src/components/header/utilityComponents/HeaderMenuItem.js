import React from "react"
import "../styles/HeaderStyles.css"

export default class HeaderMenuItem extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            hovered: false,
            containerId: "headerMenuOption" + this.props.menuItem.ID,
            textId: "headerMenuOptionText" + this.props.menuItem.ID
        };

        this.handleMouseEnter = this.handleMouseEnter.bind(this);
        this.handleMouseLeave = this.handleMouseLeave.bind(this);
    }

    componentDidMount() {
        document.getElementById(this.state.containerId).classList.toggle("headerMenuItemFadeIn");
        document.getElementById(this.state.textId).classList.toggle("menuItemTextslideIn");
    }

    render() {
        let menuOptionTextStyle = this.getMenuOptionTextStyle()

        return (
            <div id={this.state.containerId}
                className="headerMenuOption"
                onMouseEnter={this.handleMouseEnter}
                onMouseLeave={this.handleMouseLeave}>
                <div id={this.state.textId}
                    style={menuOptionTextStyle}
                    className="headerMenuOptionText">
                    {this.props.menuItem.Name}
                </div>
            </div>
        );
    }

    handleMouseEnter() {
        this.setState({
            hovered: true
        })
    };

    handleMouseLeave() {
        this.setState({
            hovered: false
        })
    };

    getMenuOptionTextStyle() {
        return {
            fontFamily: "Arial",
            color: this.state.hovered ? "#ffffff" : "#ff4500",
            fontSize: "20px",

            marginRight: "10px",

            cursor: "default",

            //forbid the text selection
            "-ms-user-select": "none",
            "-moz-user-select": "none",
            "-khtml-user-select": "none",
            "-webkit-user-select": "none"
        }
    }
}
