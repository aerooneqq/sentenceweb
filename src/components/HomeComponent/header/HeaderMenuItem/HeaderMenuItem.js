import React from "react"

//Styles
import "./HeaderMenuItemStyles.css"

export default class HeaderMenuItem extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            hovered: false,
            containerId: "headerMenuOption" + this.props.menuItem.ID,
            textId: "headerMenuOptionText" + this.props.menuItem.ID
        };

        this.handleMouseEnter = this.handleMouseEnter.bind(this);
        this.handleMouseLeave = this.handleMouseLeave.bind(this);
    }
    
    handleMouseEnter() {
        this.setState({
            hovered: true
        });
    };

    handleMouseLeave() {
        this.setState({
            hovered: false
        });
    };

    render() {
        return (
            <div id={this.state.containerId}
                 className="headerMenuOption"
                 onMouseEnter={this.handleMouseEnter}
                 onMouseLeave={this.handleMouseLeave}>
                <div id={this.state.textId}
                     className="headerMenuOptionText">
                        {this.props.menuItem.Name}
                </div>
            </div>
        );
    }
}
