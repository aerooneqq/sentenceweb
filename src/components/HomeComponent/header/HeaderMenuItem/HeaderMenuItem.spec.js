import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import HeaderMenuItem from './HeaderMenuItem';

describe("Header menu item", () => { 
    it("should render correctly", () => { 
        const props = { 
            menuItem: { 
                ID: 0,
                name: "name"
            }
        }
        const headerMenuItem = shallow(<HeaderMenuItem {...props}/>);

        expect(shallowToJson(headerMenuItem)).toMatchSnapshot();
    });

    it("should display right name", () => { 
        const props = {
            menuItem: { 
                ID: 1,
                name: "test name"
            }
        }

        const headerMenuItem = shallow(<HeaderMenuItem {...props} />);

        expect(headerMenuItem.find("div.headerMenuOptionText").text()).toBe(props.menuItem.name);
    });

    it("should change state (hover to true) when mouse enters", () => { 
        const props = { 
            menuItem: { 
                ID: 1,
                name: "Test name"
            }
        }

        const headerMenuItem = shallow(<HeaderMenuItem {...props}/>);

        headerMenuItem.find("div.headerMenuOption").simulate("mouseEnter");

        expect(headerMenuItem.state("hovered")).toBe(true);
    });

    it("should change state (hovered to false) when mouse leaves", () => { 
        const props = { 
            menuItem: { 
                ID: 1,
                name: "Test name"
            }
        }
        const headerMenuItem = shallow(<HeaderMenuItem {...props}/>);

        headerMenuItem.find("div.headerMenuOption").simulate("moueseLeave");

        expect(headerMenuItem.state("hovered")).toBe(false);
    });

    it("should have correct ids", () => { 
        const props = { 
            menuItem: { 
                ID: 1231,
                name: "Test name"
            }
        }

        const headerMenuItem = shallow(<HeaderMenuItem {...props} />);

        expect(headerMenuItem.state("containerId")).toBe("headerMenuOption" + props.menuItem.ID);
        expect(headerMenuItem.state("textId")).toBe("headerMenuOptionText" + props.menuItem.ID);
    });
});