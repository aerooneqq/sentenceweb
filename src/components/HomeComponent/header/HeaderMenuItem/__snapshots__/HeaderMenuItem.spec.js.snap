// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header menu item should render correctly 1`] = `
<div
  className="headerMenuOption"
  id="headerMenuOption0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="headerMenuOptionText"
    id="headerMenuOptionText0"
  >
    name
  </div>
</div>
`;
